src/main/java/uk/nhs/hee/trainee/details/service/PersonalDetailsService.java

```diff
@@ -26,6 +26,17 @@
 
 public interface PersonalDetailsService {
 
+  /**
+   * Update the basic details for the trainee with the given TIS ID, or create a new profile if the
+   * trainee does not exist.
+   *
+   * @param tisId           The TIS id of the trainee.
+   * @param personalDetails The personal details to add to the trainee.
+   * @return The updated or created personal details.
+   */
+  PersonalDetails createProfileOrUpdateBasicDetailsByTisId(String tisId,
```

Comment:
I've not had a chance to review the code in this repo but most of the time in tis we only have one implementation of an interface, when that happens there isn't a point in having one. Do we have multiple implementations?

Reply:
One implementation per interface, which probably is redundant.

Reply:
might be worth removing then